src/main/java/radiounit/EricssonLteRadioUnitReceiver.java

```diff
@@ -13,60 +15,193 @@
  */
 public class EricssonLteRadioUnitReceiver implements RadioUnitReceiver {
 
-	private List<Carrier> carriers;
+	/*
+	 * This is the critical resource of a radio.
+	 * 
+	 * The "volatile" keyword ensures that this resource is not cached, and is retrieved directly from
+	 * 	memory every time it is accessed.
+	 * 
+	 * The three "-Internal" methods below are all synchronized and have exclusive access to this resource. 
+	 */
+	private volatile List<Carrier> carriers;
+	
+	public EricssonLteRadioUnitReceiver()
+	{
+		this.carriers = new ArrayList<Carrier>();
+	}
 
 	public void setupEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] setupEricssonLte");
 	}
 
 	public void activateEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] activateEricssonLte");
 	}
 
 	public void deactivateEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] deactivateEricssonLte");
 	}
 
 	public void releaseEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] releaseEricssonLte");
 	}
 
-	public void setupCarrierEricssonLte(Carrier carrrier) {
-		// TODO Auto-generated method stub
-
+	public void setupCarrierEricssonLte(Carrier carrier) {
+		System.out.println("[EricssonLteRadioUnitReceiver] setupCarrierEricssonLte: " + carrier);
+		
+		// Validate carrier
+		FrequencyBand band = carrier.getCarrierFrequencies();
+		boolean isLte = false;
+		
+		for (LteFrequencyBand lteFreq : LteFrequencyBand.values())
```

Comment:
I think this makes sense. To clarify, we check the carrier frequency with the standard LTE frequencies to see if the carrier frequency is indeed a valid LTE frequency. If it isn't, an LTE frequency, we just print an error message and return.

Reply:
Ya that's correct - it's a little bit hacky, however we don't have a better way of extracting the RAT type without explicitly checking if it's an `instanceof` the `LteCarrier` class (which is also hacky).

This seems like the least hacky way that *should* guarantee we're working with an LTE carrier, however it doesn't guard against a carrier being created with the wrong frequency (e.g. a WCDMA carrier using an LTE frequency would "pass" this condition and be accepted).

I think the preferred way would be to add a `getRatType()` method to the carrier interface, and use that. Probably a discussion for the team.